### DISCUSSION TITLE
Getting revocation token

### DISCUSSION THREAD

**Discussion by Yalchin403** [walt-id/.github#41] [Help] [answered]
Hello, I wonder how can I get revocation token of a specific credential? In the official documentation (Signatory API), [here](https://docs.walt.id/v/ssikit/getting-started/rest-apis/signatory-api#revocations) I can see that endpoint to revoke specific credential expects and `id`, however I do not understand where I can get it. I also can see that documentation states that we get the token upon issuing the credential, but after sending credential issue request, I do not see the token in the response.

Thanks beforehand

**Comment by philpotisk** [comment, marked as answer]
The idea of this basic-revocation mechanism is that in the scope of credential-issuance, a token (derived from a base-token) is generated and embedded in the credentialStatus attribute of the credential. This credentialStatus field then can be checked by a verifier. 
This test-case shows the logic: https://github.com/walt-id/waltid-ssikit/blob/105796b3f9a73da3e634afc36d7525576dffe6c1/src/test/kotlin/id/walt/signatory/RevocationServiceTest.kt#L9

In case you are looking for a more advanced revocation-mechanism, please let us know. We are already working on other options as well.
